### DISCUSSION TITLE
Are styles created in the client component only available in the client component?

### DISCUSSION THREAD

**Discussion by ghost** [chakra-ui/panda#2617] [Q&A]
In the client component where you wrote use clinet, you defined the style as css.raw and exported it.

I imported the styles from another server component and merged them, but the imported styles are not applied.

However, if I change the useing component to the client component, it works again.

I've tried defining the styles in a ts file and exporting it and it works fine, so it seems to be a problem with styles created in the client component.

Is it because the client component and server component read the styles at different times?

**exported style**
/src/component/Header.tsx
![스크린샷 2024-05-21 오후 6 27 42](https://github.com/chakra-ui/panda/assets/147475576/9a0d42d3-6d47-4eb5-a2de-8bbd34c5043d)

**imported style**
/src/component/HomeOnlyHeader.tsx
![스크린샷 2024-05-21 오후 6 29 02](https://github.com/chakra-ui/panda/assets/147475576/1e0fad7f-7812-4240-9bfe-af1dd3fa5ee5)

**Comment by astahmer** [comment]
can you make a minimal reproduction ?

**Reply by ghost** [reply]
@astahmer 

https://github.com/phonego-ryong/panda-exam

I have created an example

NormalText is a client component, and we're defining and exporting styles with 
We are defining and exporting styles in CSS and CSS.RAW.

The SmallText component is the server component and is importing and using a smallTextStyle defined in css.

The RedText component is the server component and is importing and using the normalTextStyle defined in css.raw.

When I run the app, the style is not being applied to the RedText, but the SmallText is being applied normally.
However, if you change the RedText to a client component or change the NormalText component to a server component, it works fine.

**Reply by astahmer** [reply]
got it, thanks !
the issue here is Nextjs specific

since [`NormalText.tsx`](https://github.com/phonego-ryong/panda-exam/blob/main/src/components/text/NomalText.tsx) is a client component (contains `use client`), the RSC doesn't seem to be able to resolve the value of `normalTextStyle` (it logs an empty object `{}`) in the [`RedText`](https://github.com/phonego-ryong/panda-exam/blob/main/src/components/text/RedText.tsx#L7) component

now, if you move the `normalTextStyle` from `NormalText.tsx` (contains `use client`) to a `styles.ts` file, and import `normalTextStyle` from there in both `NormalText.tsx` and `RedText.tsx`, the styles will work as expected

```ts
// styles.ts
import { css } from "../../../styled-system/css";

export const normalTextStyle = css.raw({
  fontSize: "24px",
  color: "blue.400",
});

```
